app/models/queue_it/queue.rb

```diff
@@ -27,21 +27,41 @@ def two_nodes?
       size == 2
     end
 
+    def get_next_by_with_queue_length_two(nodable_attribute, attribute_value)
+      if head_node.nodable.send(nodable_attribute) == attribute_value
+        get_next_in_queue_with_length_two
+      else
+        tail_node
+      end
+    end
+
     def get_next_in_queue_with_length_two
-      next_node = ActiveRecord::Base.transaction do
+      ActiveRecord::Base.transaction do
         lock!
         old_head_node = head_node.lock!
         old_tail_node = tail_node.lock!
-        nodes.where.not(kind: :any).each { |node| node.update!(kind: :any) }
+        nodes.where.not(kind: :any).find_each { |node| node.update!(kind: :any) }
         old_head_node.update!(kind: :tail, parent_node: old_tail_node)
         old_tail_node.update!(kind: :head, parent_node: nil)
         old_head_node
       end
-      next_node
+    end
+
+    def get_next_by_in_generic_queue(nodable_attribute, attribute_value)
+      return get_next_in_queue_generic if head_node.nodable.send(nodable_attribute) == attribute_value
+
+      current_node = head_node.child_node
+      while !(current_node.nodable.send(nodable_attribute) == attribute_value && current_node != tail_node)
```

Comment:
**[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Layout/LineLength: Line is too long. [107/100]